api/lib/infrastructure/repositories/schooling-registration-repository.js

```diff
@@ -327,6 +328,20 @@ module.exports = {
       });
   },
 
+  async getUserIdByNationalStudentIdFirstNameLastNameAndBirthdate({ nationalStudentId, firstName, lastName, birthdate }) {
+    const schoolingRegistration = await knex
+      .from('schooling-registrations')
+      .where({ nationalStudentId, firstName, lastName, birthdate })
+      .whereNotNull('userId')
+      .first();
+
+    if (!schoolingRegistration) {
+      throw new NotFoundError('Not found user');
```

Comment:
**chipotage**
```suggestion
      throw new NotFoundError('User not found');
```

Reply:
Proposition de Christophe 
```suggestion
     throw new UserNotFoundError();
```